test/extensions/load_balancing_policies/subset/subset_test.cc

```diff
@@ -1706,6 +1706,7 @@ TEST_P(SubsetLoadBalancerTest, ZoneAwareFallbackAfterUpdate) {
   envoy::config::core::v3::Locality local_locality;
   local_locality.set_zone("0");
 
+  EXPECT_CALL(random_, random()).WillRepeatedly(Return(0));
```

Comment:
Would it make more sense to just use a real random generator in these tests, as opposed to one that just returns 0?

Reply:
The issue is that the same RNG is used in this test (for example `EXPECT_CALL(random_, random()).WillOnce(Return(0)).WillOnce(Return(9999)).WillOnce(Return(2));` a few lines above), and I think it needs to return the specified values in some cases to ensure that the test passes.